lib/rack/display_cycle_has_ended_page.rb

```diff
@@ -0,0 +1,29 @@
+module Rack
+  class DisplayCycleHasEndedPage
+    include Rails.application.routes.url_helpers
+
+    def initialize(app)
+      @app = app
+    end
+
+    def call(env)
+      if cycle_has_ended? && !request_to_cycle_has_ended_page?(env)
+        [301, { "Location" => cycle_has_ended_path, "Content-Type" => "text/html" }, []]
```

Comment:
This should probably be a 302, since the redirect is only temporary. This will prevent search engines from updating their index.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/302

Reply:
This middleware has been removed now